std/shared/src/main/scala/cats/effect/std/Dispatcher.scala

```diff
@@ -46,18 +48,12 @@ object Dispatcher {
   def apply[F[_]](implicit F: Async[F]): Resource[F, Dispatcher[F]] = {
     final case class Registration(action: F[Unit], prepareCancel: F[Unit] => Unit)
 
-    final case class State(end: Long, registry: LongMap[Registration]) {
-
-      /*
-       * We know we never need structural equality on this type. We do, however,
-       * perform a compare-and-swap relatively frequently, and that operation
-       * delegates to the `equals` implementation. I like having the case class
-       * for pattern matching and general convenience, so we simply override the
-       * equals/hashCode implementation to use pointer equality.
-       */
-      override def equals(that: Any) = this eq that.asInstanceOf[AnyRef]
-      override def hashCode = System.identityHashCode(this)
-    }
+    final case class State(end: Long, registry: LongMap[Registration])
+
+    sealed trait CancelState
+    case object CancelInit extends CancelState
+    final case class CancelledNoToken(promise: Promise[Unit]) extends CancelState
```

Comment:
```suggestion
    final case class CanceledNoToken(promise: Promise[Unit]) extends CancelState
```

Just to be consistent with `F.canceled`